spec/steps/approval_steps.rb

```diff
@@ -12,13 +16,24 @@ module ApprovalSteps
     @token = ApiToken.create!(user_id: @user.id, cart_id: @cart.id)
   end
 
-  step "the cart has an approval for :email" do |email|
+  step "the cart has an approval for :email in position :position" do |email, position|
     approver = User.find_or_create_by(email_address: email)
-    @cart.approvals << FactoryGirl.create(:approval, role: 'approver', user_id: approver.id)
+    @cart.approvals << FactoryGirl.create(:approval, role: 'approver', user_id: approver.id, position: position)
   end
 
   step "feature flag :flag_name is :value" do |flag, value|
     ENV[flag] = value
   end
 
+  step 'the cart has been approved by the logged in user' do
+    approval = @cart.approvals.where(user_id: @current_user.id).first
+    approval.update_attributes(status: 'approved')
```

Comment:
Yep; after you merge master, you'll have access to the workflow methods, including `approve!`